### PR TITLE
bundle

### PR DESCRIPTION
**Pull request by Zrp200**

### CODE REVIEW

File: SPD-classes/src/main/java/com/watabou/utils/Bundle.kt [hunk 1]
```diff
@@ -289,12 +289,10 @@ class Bundle private constructor(
...
             // JSONTokenizer only has a string-based constructor on Android/iOS.
-            val reader = BufferedReader(InputStreamReader(checkCompression(stream)))
-            val jsonBuilder = StringBuilder()
-            var line: String?
-            while (reader.readLine().also { line = it } != null) jsonBuilder.append(line + "\n")
-            var json = JSONTokener(jsonBuilder.toString()).nextValue()
-            reader.close()
+            var json = JSONTokener(buildString {
+                BufferedReader( InputStreamReader(checkCompression(stream)) )
+                    .forEachLine { append(it + "\n") }
```
Comment (QuasiStellar) [inline SPD-classes/src/main/java/com/watabou/utils/Bundle.kt]: formatting

File: SPD-classes/src/main/java/com/watabou/utils/Bundle.kt [hunk 2]
```diff
@@ -96,139 +96,163 @@ class Bundle private constructor(
...
+        initArray = { size, init -> Array(size, init).filterNotNull() },
+        getValue = { Bundle(getJSONObject(it)).get() }
+    )
+
+    /**
+     * Gets an array from [data] corresponding to [key].
+     * @param initArray The constructor for the array. Determines the return type of the function
+     * @param getValue The method to be used to extract each value from the retrieved [JSONArray].
+     */
+    private inline fun <R,RArray> getArray(
```
Comment (QuasiStellar) [inline SPD-classes/src/main/java/com/watabou/utils/Bundle.kt]: formatting

**Review by QuasiStellar — CHANGES_REQUESTED** [review]

File: SPD-classes/src/main/java/com/watabou/utils/Bundle.kt (hunk 2 above)
Comment (Zrp200) [reply SPD-classes/src/main/java/com/watabou/utils/Bundle.kt]: exactly what's wrong here? all of the parameters are on separate lines this time.

Comment (QuasiStellar) [reply SPD-classes/src/main/java/com/watabou/utils/Bundle.kt]: Space between type arguments (just use ctrl+alt+L)

**Review by QuasiStellar — APPROVED** [review]

**Pull request merged**